jina/clients/base/grpc.py

```diff
@@ -123,31 +188,41 @@ async def _get_results(
                 tls=self.args.tls,
                 aio_tracing_client_interceptors=self.aio_tracing_client_interceptors(),
             ) as channel:
-                stub = jina_pb2_grpc.JinaRPCStub(channel)
                 self.logger.debug(f'connected to {self.args.host}:{self.args.port}')
 
                 with ProgressBar(
                     total_length=self._inputs_length, disable=not self.show_progress
                 ) as p_bar:
                     try:
-                        async for resp in stub.Call(
-                            req_iter,
-                            compression=self.compression,
-                            metadata=metadata,
-                            credentials=kwargs.get('credentials', None),
-                            timeout=kwargs.get('timeout', None),
-                        ):
-                            callback_exec(
-                                response=resp,
-                                on_error=on_error,
-                                on_done=on_done,
-                                on_always=on_always,
-                                continue_on_error=continue_on_error,
-                                logger=self.logger,
-                            )
-                            if self.show_progress:
-                                p_bar.update()
-                            yield resp
+                        if stream:
+                            stub = jina_pb2_grpc.JinaRPCStub(channel)
```

Comment:
can't we generate the stub inside _stream_rpc ? looks like _stream_rpc will only accept JinaRPCStub anyways